### ISSUE TITLE
`set_variations` seemingly broken for certain fonts

### ISSUE THREAD

**Issue by stenson** [harfbuzz/uharfbuzz#19]
Not sure if this is a uharfbuzz-related issue (or an issue in harfbuzz itself), but on certain variable TTFs (examples below), the `Font.set_variations` method seems to have no effect on resulting `glyph_positions`.

Here’s the code I’m using (lightly modified from the example in this repo):

```python
        face = hb.Face(fontdata)
        font = hb.Font(face)
        font.scale = (face.upem, face.upem)
        hb.ot_font_set_funcs(font)
        font.set_variations(axes) # axes is a dictionary
        
        buf = hb.Buffer()
        buf.add_str(text)
        buf.guess_segment_properties()
        hb.shape(font, buf, features) # features is a dictionary
```

I should note that setting `features` in the `hb.shape` function does seem to work as expected.

Quite possibly there’s something I’m doing incorrectly, but I’ve been unable to find any examples on the internet of open-source code using the `hb_font_set_variations` function directly. Everything I’ve seen, such as [fontview](https://github.com/googlefonts/fontview) which uses libraqm, seems to instantiate variations indirectly via the `hb_ft_font_create_referenced` function (worth noting that fontview does display these variations and glyph_positions correctly for all the fonts below).

![image](https://user-images.githubusercontent.com/89770/60547326-2e949500-9cd4-11e9-8216-6003c68a5739.png)

_From top-to-bottom, [Obviously](https://www.futurefonts.xyz/ohno/obviously) (correct positions), [Mutator Sans](https://github.com/LettError/mutatorSans) (correct positions), [Vinila](https://plau.co/shop/fonts/vinila) (incorrect positions), [Compressa (Trial version)](https://www.preusstype.com/gx/compressa/compressa_55.php) (incorrect positions). All the fonts are at their max `wdth` & `wght`, and are returned from FreeType with the correct glyph shape variations. (The height of the pink bounding boxes is totally arbitrary, not a result of Harfbuzz.)_

**Comment by behdad** [comment]
First order of business is to see if this reproduces using hb-view.

**Comment by stenson** [comment]
Ah! Sorry, should’ve checked that to begin with.

hb-view seems to render everything correctly.

For instance, here’s Vinila as rendered by hb-view:

![image](https://user-images.githubusercontent.com/89770/60620580-724cd480-9d90-11e9-8bc7-d9575f48b1b6.png)

**Comment by stenson** [comment]
Not totally sure it’s relevant but seems worth mentioning — I was looking at the ttx of the fonts mentioned above in the original post, and it appears all the non-functioning variable fonts lack `HVAR` tables, while all the functioning fonts do have `HVAR` tables.

**Comment by behdad** [comment]
Ah, right.  In `hb-ot-font` (ie. when NOT using FreeType font functions) we don't use `gvar`, so font is required to have `HVAR`.  While the spec says `HVAR` is optional, I take it as defacto required.  We will one day implement `gvar`.  Not that far in the future actually, later this year.

**Comment by stenson** [comment]
Good to hear! Is there any chance of including the `hb_ft_*` functions in uharfbuzz or is that out of scope for uharfbuzz?

**Comment by behdad** [comment]
It's one function call. I don't see why not. @anthrotype

**Comment by stenson** [comment]
all seems fixed in the 0.6.0 release!
